rholang/src/main/scala/coop/rchain/rholang/interpreter/storage/implicits.scala

```diff
@@ -36,32 +34,32 @@ object implicits {
                      ListChannelWithRandom] {
 
       def get(pattern: BindPattern, data: ListChannelWithRandom)
-        : Either[OutOfPhlogistonsError.type, Option[ListChannelWithRandom]] = {
-        val (cost, resultMatch) = SpatialMatcher
+        : Either[OutOfPhlogistonsError.type, Option[ListChannelWithRandom]] =
+        SpatialMatcher
           .foldMatch(data.channels, pattern.patterns, pattern.remainder)
-          .runWithCost(CostAccount.zero)
-
-        val result = resultMatch
+          .runWithCost(CostAccount(Integer.MAX_VALUE)) // FIXME -- must come from the input args
           .map {
-            case (freeMap: FreeMap, caughtRem: Seq[Channel]) =>
-              val remainderMap = pattern.remainder match {
-                case Some(Var(FreeVar(level))) =>
-                  val flatRem: Seq[Par] = caughtRem.flatMap(
-                    chan =>
-                      chan match {
-                        case Channel(Quote(p)) => Some(p)
-                        case _                 => None
+            case (cost, resultMatch) =>
+              resultMatch
+                .map {
+                  case (freeMap: FreeMap, caughtRem: Seq[Channel]) =>
+                    val remainderMap = pattern.remainder match {
+                      case Some(Var(FreeVar(level))) =>
+                        val flatRem: Seq[Par] = caughtRem.flatMap(
+                          chan =>
+                            chan match {
+                              case Channel(Quote(p)) => Some(p)
+                              case _                 => None
```

Comment:
I think a refactor similar to this could be done here:
https://github.com/rchain/rchain/commit/d13ff6b692b20d746c5c42bb84a56fc77fe8ddd6

Won't insist though :)